### DISCUSSION TITLE
Short Orders

### DISCUSSION THREAD

**Discussion by kbs-code** [dodid/minitrade#6] [General]
1. The README says multi-asset short orders are unsupported yet as I read [this ](https://dodid.github.io/minitrade/api_backtest/#minitrade.backtest.core.backtesting.Order) it says that there is some short order functionality.  I'm assuming this is just for single-asset strategies?  Can you clarify to what extent short orders are supported?  

2. How difficult would it be to modify the source to support multi-asset short orders?

3. Why didn't you implement this functionality?  

Thanks for making this project.

**Comment by dodid** [comment]
You are right that short order support is restricted to what's available in original `Backtesting.py` and is only for single asset strategies. It's not implemented because 1) I don't have a need for it yet -), 2) I am not aware of anyone needing it, yet, and 3) it is probably not straightforward to get right. The complications include, but are not limited to: 1) how to account for short constraints like borrowing fees, margin requirements, availability, 2) how to handle margin calls, 3) how to model leverage constraint as you have positive and negative weights, 4) the interoperability between rebalance api, imperative buy/sell api and other part of the library, such as metrics calculation, plotting, 5) testing. It's not my priority for now, but if someone can come up with a neat solution, I am open to contributions.

**Reply by kbs-code** [reply]
Thanks for replying.  The use case for this would be if a trader wants to simulate trading at a firm where they would be placing long/short orders for multiple instruments.  Could I do this by running multiple single asset backtests and adding all the stats together?

For example if one was concerned about hitting a max drawdown limit while trading multiple instruments.  I don't think you could simply add multiple max drawdown stats since that wouldn't take into account when drawdowns occured.

**Reply by dodid** [reply]
I don't think adding up separate backtests works unless the buy/sell decisions for each instrument are independent and each gets fixed value allocation. If that's the case, potentially you can take the equity curve from each backtest result and recreate that for the whole portfolio, which can give you the overall drawdown. But I would doubt anyone would run a long/short portfolio that way.
